armi/settings/settingsValidation.py

```diff
@@ -346,6 +346,7 @@ def _inspectSettings(self):
             CONF_XS_KERNEL,
             CONF_XS_SCATTERING_ORDER,
         )
+        from armi.settings.fwSettings.globalSettings import CONF_ZONE_DEFINITIONS, CONF_ZONES_FILE
```

Comment:
This can't be at the top of the file because it causes an import loop?

Reply:
That's correct.

This particular module is a bit wonky. It serves both to define the machinery which allows the settings validation, and then also proceeds to use that machinery and do a bunch of settings validation. But doing the settings validation from within that module is an anti-pattern, because it requires a bunch of circular imports due to the fact that the actual settings are defined in other modules.

Ideally, the settings validation would be moved to the associated modules which define the settings themselves.